unittests/test_utility.py

```diff
@@ -979,19 +979,19 @@ def test_reversed(self):
 
     def test_concat_files(self):
         tmpdir = tempfile.mkdtemp(dir='unittests')
```

Comment:
Instead of creating it and deleting it manually, consider using [tempfile.TemporaryDirectory](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory), which is also a context manager.